gateway-js/src/loadSupergraphSdlFromStorage.ts

```diff
@@ -36,28 +43,74 @@ const { name, version } = require('../package.json');
 
 const fetchErrorMsg = "An error occurred while fetching your schema from Apollo: ";
 
-export async function loadSupergraphSdlFromStorage({
-  graphId,
-  graphVariant,
-  apiKey,
-  endpoint,
+async function submitOutOfBandReport({
+  error,
+  request,
+  response,
+  startedAt = '',
+  endedAt = '',
+  tags,
   fetcher,
 }: {
-  graphId: string;
-  graphVariant: string;
-  apiKey: string;
-  endpoint: string;
+  error: any;
+  request: Request;
+  response?: Response;
+  startedAt: string;
+  endedAt: string;
+  tags?: string[];
   fetcher: typeof fetch;
 }) {
-  let result: Response;
+
+  // don't send report if the endpoint url is not configured
+  if (!process.env.APOLLO_OUT_OF_BAND_REPORTER_ENDPOINT) {
+    return;
+  }
+
+  let errorCode = ErrorCode.Other;
+
+  // some possible error situations to check against
+  if (response?.status && [400, 413, 422].includes(response?.status)) {
+    errorCode = ErrorCode.InvalidBody;
+  }
+  else if (response?.status && [408, 504].includes(response?.status)) {
+    errorCode = ErrorCode.Timeout;
+  }
+  else if (!response || [502, 503].includes(response?.status)) {
+    errorCode = ErrorCode.ConnectionFailed;
+  }
+
   try {
-    result = await fetcher(endpoint, {
+    let responseBody;
+    try{
+      responseBody = await response?.json();
+    } catch (e) {
+      responseBody = null;
+    }
+
+    const result = await fetcher(process.env.APOLLO_OUT_OF_BAND_REPORTER_ENDPOINT!!, {
       method: 'POST',
       body: JSON.stringify({
-        query: SUPERGRAPH_SDL_QUERY,
+        query: OUT_OF_BAND_REPORTER_QUERY,
         variables: {
-          ref: `${graphId}@${graphVariant}`,
-          apiKey,
+          input: {
+            error: {
+              code: errorCode,
+              message: error.message ?? error
+            },
+            request: {
+              url: request.url,
+              headers: Object.entries(request.headers),
+              body: request.bodyUsed ? await request.json() : ''
+            },
+            response: response ? {
+              httpStatusCode: response.status,
+              headers: Object.entries(response.headers),
```

Comment:
Similar, but lessened concern as the `request` stuff.  Perhaps just the status code could would capture what we want and limit the possibility of us re-transmitting 🏓 
 other response `headers` we might get back?